internal/controller/util/json_util.go

```diff
@@ -0,0 +1,348 @@
+			return false, fmt.Errorf("operands not supported for operator: %v, %v, %s",
+				a.Kind(), b.Kind(), operator)
+		}
+	}
+
+	// Here, we have two scenarios:
+	// 1. operands are either string or float64 and operator is any of the 6
+	// 2. operands are of any kind and operator is either == or !=
+
+	// Safety latch:
+	// We will initially support only string, float64 and bool
+	// return an error if we encounter any other kind
+	if isUnsupportedKind(a.Kind()) || isUnsupportedKind(b.Kind()) {
+		return false, fmt.Errorf("unsupported kind for comparison: %v, %v", a.Kind(), b.Kind())
+	}
+
+	if isKindBool(a.Kind()) && isKindBool(b.Kind()) {
+		return compareBool(a.Bool(), b.Bool(), operator)
+	}
+
+	return compareValues(a.Interface(), b.Interface(), operator)
+}
+
+func compareBool(a, b bool, operator string) (bool, error) {
+	switch operator {
+	case "==":
+		return a == b, nil
+	case "!=":
+		return a != b, nil
+	default:
+		return false, fmt.Errorf("unknown operator: %s", operator)
+	}
+}
+
+func compareValues(val1, val2 interface{}, operator string) (bool, error) {
+	switch v1 := val1.(type) {
+	case float64:
+		v2, ok := val2.(float64)
+		if !ok {
+			return false, fmt.Errorf("mismatched types")
+		}
+		switch operator {
+		case "==":
+			return v1 == v2, nil
+		case "!=":
+			return v1 != v2, nil
+		case "<":
+			return v1 < v2, nil
+		case ">":
+			return v1 > v2, nil
+		case "<=":
+			return v1 <= v2, nil
+		case ">=":
+			return v1 >= v2, nil
+		}
+	case string:
+		v2, ok := val2.(string)
+		if !ok {
+			return false, fmt.Errorf("mismatched types")
+		}
+		switch operator {
+		case "==":
+			return v1 == v2, nil
+		case "!=":
+			return v1 != v2, nil
+		}
+	case bool:
+		v2, ok := val2.(bool)
+		if !ok {
+			return false, fmt.Errorf("mismatched types")
+		}
+		switch operator {
+		case "==":
+			return v1 == v2, nil
+		case "!=":
+			return v1 != v2, nil
+		}
+	}
+	return false, fmt.Errorf("unsupported type or operator")
+}
+
+func isKindString(kind reflect.Kind) bool {
+	return kind == reflect.String
+}
+
+func isKindFloat64(kind reflect.Kind) bool {
+	return kind == reflect.Float64
+}
+
+func isKindBool(kind reflect.Kind) bool {
+	return kind == reflect.Bool
+}
+
+func isUnsupportedKind(kind reflect.Kind) bool {
+	return kind != reflect.String && kind != reflect.Float64 && kind != reflect.Bool
+}
+
+func isKindStringOrFloat64(kind reflect.Kind) bool {
+	return isKindString(kind) || isKindFloat64(kind)
+}
+
+func parseBooleanExpression(booleanExpression string) (op string, jsonPaths []string, err error) {
+	operators := []string{"==", "!=", ">=", ">", "<=", "<"}
+
+	// TODO
+	// If one of the jsonpaths have the operator that we are looking for,
+	// then strings.Split with split it at that point and not in the middle.
+
+	for _, op := range operators {
+		exprs := strings.Split(booleanExpression, op)
+
+		jsonPaths = trimLeadingTrailingWhiteSpace(exprs)
+
+		if len(exprs) == 2 &&
```

Comment:
I prefer not to use hardcoded numbers in code.
It's better to use something like: 
const ExpectedExpressionCount = 2
if len(exprs) == ExpectedExpressionCount